### PR TITLE
Add ASR tarred dataset documentation

### PR DESCRIPTION
**Pull request by redoctopus**
Signed-off-by: Jocelyn Huang <jocelynh@nvidia.com>

### CODE REVIEW

File: docs/source/asr/datasets.rst [hunk 1]
```diff
@@ -190,3 +192,81 @@ in the same directory as the manifest, or even in any specific directory structu
...
+Then, you will need to pass in the paths to all of your audio tarballs in ``tarred_audio_filepaths``, either as a list
+of filepaths (e.g. ``['/data/shard1.tar', '/data/shard2.tar']``), or in a single brace-expandable string (e.g.
+``'/data/shard_{1..64}.tar'``).
+
+.. note::
+  For brace expansion, there may be cases where ``{x..y}`` syntax cannot be used due to shell interference.
+  This occurs most commonly inside SLURM scripts. Therefore we provide a few equivalent replacements.
+  Supported opening braces (equivalent to ``{``) are ``(``, ``[``, ``<`` and the special tag ``_OP_``.
+  Supported closing braces (equivalent to ``}``) are ``)``, ``]``, ``>`` and the special tag ``_CL_``.
+  For SLURM based tasks, we suggest the use of the special tags for ease of use.
```
Comment (titu1994) [inline docs/source/asr/datasets.rst:217]: Lets guide users to user the `_OP_` and `_CL_` tags - they are guarenteed to work. The others may or may not work depending on the shell they use (bash vs zsh has different reactions to < vs [ )

File: docs/source/asr/datasets.rst [hunk 2]
```diff
@@ -190,3 +192,81 @@ in the same directory as the manifest, or even in any specific directory structu
...
+
+As with non-tarred datasets, the manifest file should be passed in ``manifest_filepath``.
+The dataloader will assume that the length of the manifest after filtering is the correct size of the dataset
+for reporting training progress.
+
+The ``tarred_shard_strategy`` field of the config file can be set if you have multiple shards and are running
+an experiment with multiprocessing.
```
Comment (titu1994) [inline docs/source/asr/datasets.rst:224]: Should this be `multinode` instead of `multiprocessing`?

File: docs/source/asr/datasets.rst [hunk 3]
```diff
@@ -190,3 +192,81 @@ in the same directory as the manifest, or even in any specific directory structu
...
+It defaults to ``scatter``, which preallocates a set of shards per node which do not change during runtime.
+
+For more information about the individual tarred datasets and the parameters you can set, including shuffling options,
+see the corresponding class APIs on the `Datasets <./api.html#Datasets>`__ page.
+
+.. warning::
+  If using multiple processes the number of shards should be divisible by the number of workers to ensure an even
+  split among workers.
+  If it is not divisible, logging will give a warning but training will proceed, but likely hang at the last epoch.
+  In addition, if using mutiprocessing, each shard must have the same number of entries after filtering is applied.
```
Comment (titu1994) [inline docs/source/asr/datasets.rst:234]: `multinode` vs `multiprocessing`? Or does it even occur under 1 node and multiprocessing. Thats new..

**Review by titu1994 — APPROVED** [review]
Fantastic ! @fayejf Please add equivalent section for Classification models

File: docs/source/asr/datasets.rst (hunk 3 above)
Comment (redoctopus) [reply docs/source/asr/datasets.rst:234]: In the past I did get it to hang on my local machine with two GPUs (as long as world_size > 1). Though I may be mixing up my PyTorch/Python terminology here... Distributed processing?

File: docs/source/asr/datasets.rst (hunk 1 above)
Comment (redoctopus) [reply docs/source/asr/datasets.rst:217]: Sounds good to me, will update this.

**Pull request merged**